src/tests/ftest/aggregation/multiple_pool_cont.py

```diff
@@ -0,0 +1,195 @@
+        self.free_space_dict[pool][2] = val [After 2nd IOR free space]
+        self.free_space_dict[pool][3] = val [After aggregation complete]
+
+        Args:
+            storage_index (int): index of the pool free space to obtain
+        """
+        for pool in self.pool:
+            pool.get_info()
+            value = pool.info.pi_space.ps_space.s_free[storage_index]
+            if space_tag == 0:
+                self.free_space_dict[pool.uuid] = {space_tag: value}
+            else:
+                self.free_space_dict[pool.uuid][space_tag] = value
+            self.log.info("Free Space Information")
+            self.log.info("======================")
+            self.log.info(self.free_space_dict)
+
+    def test_free_space(self, space_tag1, space_tag2):
+        """Test whether the free space is decremented between two IOR runs.
+           Asserts if free sapce with space_tag1 less than free space with
+           space_tag2.
+
+        Args:
+            space_tag1 : IOR run tag (0,1,2)
+            space_tag2 : IOR run (1,2,3)
+        """
+        for pool in self.pool:
+            space_used_by_ior = (self.free_space_dict[pool.uuid][space_tag1] -
+                                 self.free_space_dict[pool.uuid][space_tag2])
+            self.log.info("Pool %s Space used by ior = %s", pool.uuid,
+                          space_used_by_ior)
+            self.assertTrue(self.free_space_dict[pool.uuid][space_tag1] >
+                            self.free_space_dict[pool.uuid][space_tag2],
+                            "IOR run was not successful.")
+
+    def test_longrun_aggregation(self, total_pools=1, total_containers_per_pool=1):
+        """Jira ID: DAOS-7326
+        Test Description:
+            This is a common method which run IOR on the specified pools,
+            container quanties and test aggregation.
+        """
+        # test params
+        total_runtime = self.params.get("total_runtime",
+                                        '/run/runtime/*')
+        pool_qty = total_pools
+        container_qty = total_containers_per_pool
+        start_time = 0
+        finish_time = 0
+
+        job_manager = get_job_manager(self, "Mpirun", None, False, "mpich",
+                                      self.get_remaining_time())
+        # Create requested pools
+        self.add_pool_qty(pool_qty, connect=False)
+        start_time = time.time()
+        while int(finish_time - start_time) < int(total_runtime):
+            # Since the transfer size is 1M, the objects will be inserted
+            # directly into NVMe and hence storage_index = 1
+            storage_index = 1
+
+            self.save_free_space(0, storage_index)
+
+            # Disable the aggregation
+            for pool in self.pool:
+                pool.set_property("reclaim", "disabled")
+
+            # Create the containers requested per pool
+            for pool in self.pool:
+                self.add_container_qty(container_qty, pool)
```

Comment:
```suggestion
            for pool in self.pool:
                pool.set_property("reclaim", "disabled")
                self.add_container_qty(container_qty, pool)
```
Can't we merge these loops?

Reply:
Done.